### PR TITLE
add checks to user update processing

### PR DESCRIPTION
**Pull request by mattkasun**
## Describe your changes
also addresses GRA-1480
## Checklist before requesting a review
- [ ] My changes affect only 10 files or less.
- [ ] I have performed a self-review of my code and tested it.
- [ ] If it is a new feature, I have added thorough tests, my code is <= 1450 lines.
- [ ] If it is a bugfix, my code is <= 200 lines.
- [ ] My functions are <= 80 lines.
- [ ] I have had my code reviewed by a peer.
- [ ] My unit tests pass locally.
- [ ] Netmaker is awesome.

### CODE REVIEW

File: logic/jwts.go [hunk 1]
```diff
@@ -101,6 +102,18 @@ func CreateUserJWT(username string, networks []string, isadmin bool) (response s
 	return "", err
 }
 
+// VerifyJWT verifies Auth Header
```
Comment (0xdcarns) [inline logic/jwts.go:105]: either comment or func name is incorrect

**Review by 0xdcarns — CHANGES_REQUESTED** [review]

File: logic/jwts.go (hunk 1 above)
Comment (mattkasun) [reply logic/jwts.go:105]: updated func name

File: controllers/user.go [hunk 2]
```diff
@@ -331,7 +331,18 @@ func updateUser(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Type", "application/json")
 	var params = mux.Vars(r)
 	// start here
+	jwtUser, _, isadmin, err := logic.VerifyJWT(r.Header.Get("Authorization"))
+	if err != nil {
```
Comment (abhishek9686) [inline controllers/user.go:335]: can't permissions be verified at middleware?

**Comment by theguy951357** [comment]
will wait on changes

File: controllers/user.go (hunk 2 above)
Comment (mattkasun) [reply controllers/user.go:335]: don't see the point of creating new middleware for a single endpoint.

**Review by theguy951357 — APPROVED** [review]
non-admin user cannot change the credentials of an admin user.
critical list passes

**Review by 0xdcarns — APPROVED** [review]

**Pull request merged**